src/main/scala/org/broadinstitute/dsde/workbench/sam/config/AppConfig.scala

```diff
@@ -44,7 +44,12 @@ object AppConfig {
       val uqPath = unquoteAndEscape(path)
       ResourceRole(
         ResourceRoleName(uqPath),
-        config.as[Set[String]](s"$uqPath.roleActions").map(ResourceAction.apply)
+        config.as[Set[String]](s"$uqPath.roleActions").map(ResourceAction.apply),
+        config.as[Option[Set[String]]](s"$uqPath.includedRoles").getOrElse(Set.empty).map(ResourceRoleName.apply),
+        config.as[Option[Map[String, Set[String]]]](s"$uqPath.descendantRoles").getOrElse(Map.empty)
+          .map { case (resourceTypeName, roleNames) =>
+            (ResourceTypeName(resourceTypeName), roleNames.map(ResourceRoleName.apply))
+          }
```

Comment:
is there a reason why the ValueReaders extend `ValueReader` instead of using `ValueReader.relative` like `directoryConfigReader` below? I think we could get rid of the `uqPath` stuff.

Reply:
The `ValueReader.relative` stuff knows the config path as opposed to the readers we have here where the config path defines the role and resource type names. I'd guess that there's some way to use the `ValueReader.relative` instead of extending `ValueReader` and still figure out the path, but it's not immediately apparent to me. From what I've seen, it doesn't seem like we could necessarily get rid of `uqPath` either since we'd still need to unquote and escape whatever path we were able to glean from the `Config` object before constructing our case classes.